components/OfficerMap.js

```diff
@@ -0,0 +1,994 @@
+    this.getGoogleMaps().then(google => {
+      clustererArray.forEach(clusterer => clusterer.clearMarkers());
+
+      const countyRE = new RegExp('county');
+      const stateRE = new RegExp('state');
+      const fedRE = new RegExp('federal');
+      const airportRE = new RegExp('airport');
+
+      const markersArray = [[], [], [], [], []];
+      data.map(function(row) {
+        const geometry = row.Geolocation.split(',');
+        const location = { lat: parseFloat(geometry[0]), lng: parseFloat(geometry[1]) };
+        const marker = new google.maps.Marker({
+          position: location,
+          info: {
+            name: row.Name,
+            dod: row.DateofDeath,
+            age: row.Age,
+            facility: row.Agency,
+            facilityType: row.AgencyType,
+            city: row.City,
+            county: row.County,
+          },
+        });
+
+        // Split date string to translate from format MM/DD/YYYY to array [MM, DD, YYYY]
+        const dateArray = row.DateofDeath.split('/');
+
+        // Input date using 'new Date(YYYY, MM, DD);'
+        // Months are in range 0-11
+        const markerDate = new Date(dateArray[2], dateArray[0] - 1, dateArray[1]);
+        if (markerDate <= date) {
+          if (selectedOption === 'all') {
+            markersArray[0].push(marker);
+          } else if (selectedOption === 'facility') {
+            if (countyRE.test(row.AgencyType)) {
+              markersArray[0].push(marker);
+            } else if (stateRE.test(row.AgencyType)) {
+              markersArray[1].push(marker);
+            } else if (fedRE.test(row.AgencyType)) {
+              markersArray[2].push(marker);
+            } else if (airportRE.test(row.AgencyType)) {
+              markersArray[3].push(marker);
+            }
+          } else if (selectedOption === 'age') {
+            if (parseInt(row.Age) < 45) {
+              markersArray[0].push(marker);
+            } else if (parseInt(row.Age) < 65) {
+              markersArray[1].push(marker);
+            } else {
+              markersArray[2].push(marker);
+            }
+          }
+        }
+        return markersArray;
+      });
+
+      for (let i = 0; i < 4; i += 1) {
+        clustererArray[i].addMarkers(markersArray[i]);
+      }
+
+      this.setState({
+        map,
+        clustererArray,
+        selectUpdate: false,
+      });
+    });
+  }
+
+  handleOptionChange(event) {
+    const { infowindow } = this.state;
+    let firstLegendText = '';
+    let secondLegendText = '';
+    let thirdLegendText = '';
+    let fourthLegendText = '';
+    if (infowindow) {
+      infowindow.close();
+    }
+    const selectedOption = event.target.value;
+    if (selectedOption === 'all') {
+      firstLegendText = 'All Deaths';
+    } else if (selectedOption === 'facility') {
+      firstLegendText = 'County';
+      secondLegendText = 'State';
+      thirdLegendText = 'Federal';
+      fourthLegendText = 'Airport';
```

Comment:
@EvaRuthM mentioned that we may want to add more facilities in the future. Could we generate the facilities from the data in the spreadsheet?

Reply:
The two could use a common object. This would require pulling up several fields to be inputs into the object itself, off the top of my head the number of classifications (ie. the different flavors of facilities/agencies), a string to represent each classifications, a color for each classifications cluster element/legend icon, and a regular expression for each classification. You could also generate them from the data in the spread sheet.